Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -179,409 +135,356 @@ public void Dispose()
+                        _lanSubnets.Add(new IPNetwork(IPAddress.Parse("fc00::"), 7)); // RFC 4193 (Unique local)
+                    }
 
-            return result;
+                    if (IsIpv4Enabled)
+                    {
+                        _lanSubnets.Add(new IPNetwork(IPAddress.Loopback, 8)); // RFC 5735 (Loopback)
+                        _lanSubnets.Add(new IPNetwork(IPAddress.Parse("10.0.0.0"), 8)); // RFC 1918 (private)
+                        _lanSubnets.Add(new IPNetwork(IPAddress.Parse("172.16.0.0"), 12)); // RFC 1918 (private)
+                        _lanSubnets.Add(new IPNetwork(IPAddress.Parse("192.168.0.0"), 16)); // RFC 1918 (private)
+                    }
+                }
+
+                _logger.LogInformation("Defined LAN addresses: {0}", _lanSubnets.Select(s => s.Prefix + "/" + s.PrefixLength));
+                _logger.LogInformation("Defined LAN exclusions: {0}", _excludedSubnets.Select(s => s.Prefix + "/" + s.PrefixLength));
+                _logger.LogInformation("Using LAN addresses: {0}", _lanSubnets.Where(s => !_excludedSubnets.Contains(s)).Select(s => s.Prefix + "/" + s.PrefixLength));
+            }
         }
 
-        /// <inheritdoc/>
-        public string GetBindInterface(IPObject source, out int? port)
+        /// <summary>
+        /// Enforce bind addresses and exclusions on available interfaces.
+        /// </summary>
+        private void EnforceBindSettings(NetworkConfiguration config)
         {
-            port = null;
-            ArgumentNullException.ThrowIfNull(source);
-
-            // Do we have a source?
-            bool haveSource = !source.Address.Equals(IPAddress.None);
-            bool isExternal = false;
-
-            if (haveSource)
+            lock (_initLock)
             {
-                if (!IsIP6Enabled && source.AddressFamily == AddressFamily.InterNetworkV6)
+                // Respect explicit bind addresses
+                var localNetworkAddresses = config.LocalNetworkAddresses;
+                if (localNetworkAddresses.Length > 0 && !string.IsNullOrWhiteSpace(localNetworkAddresses.First()))
                 {
-                    _logger.LogWarning("IPv6 is disabled in Jellyfin, but enabled in the OS. This may affect how the interface is selected.");
+                    var bindAddresses = localNetworkAddresses.Select(p => NetworkExtensions.TryParseToSubnet(p, out var network)
+                        ? network.Prefix
+                        : (_interfaces.Where(x => x.Name.Equals(p, StringComparison.OrdinalIgnoreCase))
+                            .Select(x => x.Address)
+                            .FirstOrDefault() ?? IPAddress.None))
+                        .ToList();
+                    bindAddresses.RemoveAll(x => x == IPAddress.None);
+                    _interfaces = _interfaces.Where(x => bindAddresses.Contains(x.Address)).ToList();
```

Comment:
I'm not a fan of updating the field piecemeal. I would prefer if it was an array and that it was replaced rather than built gradually. Unless we actually need it as a mutable List?